packages/dx-react-grid/src/components/table-layout/virtual-table-utils.js

```diff
@@ -14,21 +14,25 @@ export const getVisibleRows = (rows, viewportTop, viewportHeight, getRowHeight)
   let position = 0;
   for (let i = 0; i < rows.length; i += 1) {
     const row = rows[i];
-    const last = result[result.length - 1];
+    const lastIndex = result.length - 1;
+    const last = result[lastIndex];
 
     const height = getRowHeight(row);
     const nextPosition = position + height;
     if (
-      (viewportTop <= position && position < bottom
-        && viewportTop < nextPosition && nextPosition <= bottom) ||
-      (viewportTop > position && nextPosition > bottom)
+      (position >= viewportTop && position < bottom) ||
```

Comment:
Can we rename `bottom` to `viewportBottom`? It makes this condition more readable.